File: Python/Task02/n15787.py
# https://inf.reshuege.ru/problem?id=15787

print('x y z w f')
for x in [0, 1]:
    for y in [0, 1]:
        for z in [0, 1]:
            for w in [0, 1]:
                f = ((x <= y) and (y <= w)) or (z == (x or y)) 
                if not f:
                    print(x, y, z, w, int(f))

File: C++/Task02/n15787/n15787.cpp
// https://inf.reshuege.ru/problem?id=15787

#include <iostream>

int main()
{
	std::cout << "x y z w f" << std::endl;
	for (bool x : { false, true })
	{
		for (bool y : { false, true })
		{
			for (bool z : { false, true })
			{
				for (bool w : { false, true })
				{
					bool f = ((x <= y) && (y <= w)) || (z == (x || y));
					if (f == false)
					{
						std::cout << x << " " << y << " " << z << " " << w << " " << f << std::endl;
					}
				}
			}
		}
	}
	return 0;
}
